tests/drivers/eeprom/src/main.c

```diff
@@ -43,27 +43,148 @@ static void test_write_and_verify(void)
 
-	rc = eeprom_read(eeprom, 0, rd_buf, sizeof(rd_buf));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		rc = eeprom_read(eeprom, address, rd_buf, sizeof(rd_buf));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
 
-	rc = memcmp(wr_buf1, rd_buf, sizeof(wr_buf1));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		rc = memcmp(wr_buf1, rd_buf, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
 
-	rc = eeprom_write(eeprom, 0, wr_buf2, sizeof(wr_buf2));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		address += sizeof(wr_buf1);
+	}
 
-	rc = eeprom_read(eeprom, 0, rd_buf, sizeof(rd_buf));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+	address = 0;
+	while (address < size) {
+		rc = eeprom_write(eeprom, address, wr_buf2, sizeof(wr_buf2));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
 
-	rc = memcmp(wr_buf2, rd_buf, sizeof(wr_buf2));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		rc = eeprom_read(eeprom, address, rd_buf, sizeof(rd_buf));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = memcmp(wr_buf2, rd_buf, sizeof(wr_buf2));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		address += sizeof(wr_buf2);
+	}
+
+	address = 0;
+	while (address < size) {
+		rc = eeprom_write(eeprom, address, wr_buf1, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = eeprom_read(eeprom, address, rd_buf, sizeof(rd_buf));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = memcmp(wr_buf1, rd_buf, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		address += sizeof(wr_buf1);
+	}
+
+	address = 0;
+	while (address < size) {
+		rc = eeprom_write(eeprom, address, wr_buf2, sizeof(wr_buf2));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = eeprom_read(eeprom, address, rd_buf, sizeof(rd_buf));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = memcmp(wr_buf2, rd_buf, sizeof(wr_buf2));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		address += sizeof(wr_buf2);
+	}
+}
+
+static void test_constant_address(void)
+{
+	const uint8_t wr_buf1[4] = { 0xFF, 0xEE, 0xDD, 0xCC };
+	uint8_t rd_buf[sizeof(wr_buf1)];
+	const struct device *eeprom;
+	size_t size;
+	off_t address;
+	int rc;
+
+	eeprom = device_get_binding(eeprom_label);
+	size = eeprom_get_size(eeprom);
+
+	address = 0;
+	for (int cnt = 0; cnt < 16; cnt++) {
+
+		rc = eeprom_write(eeprom, address, wr_buf1, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = eeprom_read(eeprom, address, rd_buf, sizeof(rd_buf));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+		rc = memcmp(wr_buf1, rd_buf, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+
+	}
+}
+
+static void test_write_and_verify_byte(void)
+{
+	const uint8_t wr_buf1[1] = { 0x00 };
+	uint8_t rd_buf[sizeof(wr_buf1)];
```

Comment:
Since read and write size are the same, how about

const size = 1;

and then use that instead of sizeof()?

Reply:
done